### ISSUE TITLE
Could not validate provider configuration occurs when project == provider

### ISSUE THREAD

**Issue by nesl247** [pulumi/pulumi#3546]
When the provider name and the project name are the same, you'll get a `could not validate provider configuration: 1 error occurred: * : invalid or unknown key: <custom key>`.

This means that if the project and provider are the same, you cannot use the default config project namespace.

**Comment by pgavlin** [comment]
If I understand correctly, this means that you cannot have a project named "aws" that uses the AWS provider. Is that right?

**Comment by nesl247** [comment]
In this case it's `vault` that uses the `vault` provider, but yes.

**Comment by MrSimonEmms** [comment]
I've got similar - I've got a project called `gitlab` and using the GitLab provider.

In truth, I don't mind renaming my project rather than Pulumi being fixed but it would be useful to have a better error message. Suggest something along the lines of `could not validate provider configuration: custom config ${customNamespace} namespace clashes with ${providerNamespace} provider namespace`.

Could also put a link to this issue ID until it's fixed

**Comment by nesl247** [comment]
I don't think we should have to rename the project to be honest. It makes perfect sense to have the project name be the same.

What I think would make more sense is that the `project` key shouldn't be the project name, but `project`. So instead of `gitlab:key1` it would be `project:key1`.

Then this wouldn't be an issue.

**Comment by lukehoban** [comment]
> What I think would make more sense is that the project key shouldn't be the project name, but project. So instead of gitlab:key1 it would be project:key1.

FWIW - An alternative would be to truly treat un-namespaced names as being un-namespaced, instead of silently projecting them into the "project" namespace.

**Comment by nesl247** [comment]
So what you mean is that instead of `projectName:key: value` we'd just have `key`?

If so, then I'm not sure if I like it either, as it makes it harder to group all of the project specific variables together. It also has room for causing conflicts if the key name conflicted with a namespace no?

**Comment by DavidHe1127** [comment]
Have similar error, see below.

```
error: Missing required configuration variable 'auth0:custom_tld'
        please set a value using the command `pulumi config set auth0:custom_tld <value>`
```

If I run `pulumi config set auth0:custom_tld xxx`, I will get  
```
error occurred: * : invalid or unknown key: custom_tld
```

My project

```yml
# Pulumi.yaml
name: authzero-configuration
runtime: nodejs
main: src/
description: auth0 configuration as code
```

My provider's name is `Pulumi.auth0-configuration-dev.yaml`.  Any idea how I can fix it?

**Comment by DavidHe1127** [comment]
Issue was solved by renaming the namespace I put in front of my custom configs.

```yml
config:
  # not ok
  auth0:custom_config: dave.dev
  # ok
  auth0-configuration:custom_config: dave.dev
```

**Comment by markfickett** [comment]
I also found the initial error very confusing, and like MrSimonEmms' suggestion.

Setting a custom arbitrary namespace is an OK workaround for me.

```yaml
config:
  td:some_value: foobar
```

Set via: `pulumi config set td:some_value MY_VALUE` .

I was also confused by where the `Config` introduces the default namespace. I expected `Config.require("td:some_value")` to work just based on the `pulumi config set` command, but instead needed to do `Config("td").require("some_value")`.

**Comment by alechp** [comment]
One thing that wasn't immediately clear to me: the namespace collision occurs when both the project name is the same as the provider *and* it's being used in the config file

### Reproduction

1. `Deploy` - initial error
<img width="626" alt="Screen Shot 2022-10-10 at 4 59 16 PM" src="https://user-images.githubusercontent.com/3058939/194969746-857ca4c8-032f-49fe-8fcc-ffc1427ad7f4.png">

2. `Comment out Pulumi.*.yaml`
<img width="427" alt="Screen Shot 2022-10-10 at 4 59 35 PM" src="https://user-images.githubusercontent.com/3058939/194969862-948f84a4-01da-478d-8668-b42185f51854.png">

3. `Re-trigger deploy` - no errors
<img width="802" alt="Screen Shot 2022-10-10 at 4 59 23 PM" src="https://user-images.githubusercontent.com/3058939/194969890-cb350d84-3d1c-41f9-8b9c-900a5216adb9.png">
